lite/tests/classification/test_counts.py

```diff
@@ -0,0 +1,862 @@
+    for m in actual_metrics:
+        assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in actual_metrics
+
+
+def test_counts_unit(
+    classifications_from_api_unit_tests: list[Classification],
+):
+
+    loader = DataLoader()
+    loader.add_data(classifications_from_api_unit_tests)
+    evaluator = loader.finalize()
+
+    metrics = evaluator.evaluate(score_thresholds=[0.5])
+
+    # test Counts
+    actual_metrics = [m.to_dict() for m in metrics[MetricType.Counts]]
+    expected_metrics = [
+        {
+            "type": "Counts",
+            "value": {
+                "tp": [1],
+                "fp": [0],
+                "fn": [2],
+                "tn": [3],
+            },
+            "parameters": {
+                "score_thresholds": [0.5],
+                "hardmax": True,
+                "label": {"key": "class", "value": "0"},
+            },
+        },
+        {
+            "type": "Counts",
+            "value": {
+                "tp": [1],
+                "fp": [3],
+                "fn": [0],
+                "tn": [2],
+            },
+            "parameters": {
+                "score_thresholds": [0.5],
+                "hardmax": True,
+                "label": {"key": "class", "value": "1"},
+            },
+        },
+        {
+            "type": "Counts",
+            "value": {
+                "tp": [0],
+                "fp": [1],
+                "fn": [2],
+                "tn": [3],
+            },
+            "parameters": {
+                "score_thresholds": [0.5],
+                "hardmax": True,
+                "label": {"key": "class", "value": "2"},
+            },
+        },
+    ]
+    for m in actual_metrics:
+        assert m in expected_metrics
+    for m in expected_metrics:
+        assert m in actual_metrics
+
+
+def test_counts_with_example(
+    classifications_two_categeories: list[Classification],
+):
+
+    loader = DataLoader()
+    loader.add_data(classifications_two_categeories)
+    evaluator = loader.finalize()
+
+    metrics = evaluator.evaluate(score_thresholds=[0.05, 0.5, 0.95])
+
+    # test Counts
+    actual_metrics = [m.to_dict() for m in metrics[MetricType.Counts]]
+    expected_metrics = [
+        {
+            "type": "Counts",
+            "value": {
+                "tp": [1, 1, 0],
+                "fp": [0, 0, 0],
+                "fn": [2, 2, 3],
+                "tn": [3, 3, 3],
+            },
+            "parameters": {
+                "score_thresholds": [0.05, 0.5, 0.95],
+                "hardmax": True,
+                "label": {"key": "animal", "value": "bird"},
+            },
+        },
+        {
+            "type": "Counts",
+            "value": {
+                "tp": [
+                    0,
+                    0,
+                    0,
+                ],  # TODO shouldn't the sixth groundtruth be a tp at low score thresholds? the groundtruth is "dog", and both "cat" and "dog" share a score of .4
```

Comment:
I expanded this test a bit to include more score thresholds and added comments in the form of "TODO"s

Reply:
The existing results are correct for the input args. For the regular Counts metric in this test we are taking the hardmax. This means that one prediction "wins" and the others are considered to be false.

Reply:
There is no winner here b/c the score_threshold is `0.5` and the highest score is `0.4`

Reply:
yes, I asked you to include the score threshold of 0.05 to check for this (which I had added, and you later removed)

Reply:
This is included now!